### PR TITLE
remove unnecessary  string operation

### PR DESCRIPTION
**Pull request by gaozhangmin**
### Motivation

Some string operation is not necessary

### Modifications

remove unnecessary string operation

### Documentation

Check the box below and label this PR (if you have committer privilege).

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
code optimize
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

**Review by Shoothzj — APPROVED** [review]

**Review by codelipenghui — APPROVED** [review]

File: pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/BaseResource.java [hunk 1]
```diff
@@ -267,7 +267,7 @@ public PulsarAdminException getApiException(Response response) {
 
     public static String getReasonFromServer(WebApplicationException e) {
         try {
-            return e.getResponse().readEntity(ErrorData.class).reason.toString();
```
Comment (RobertIndie) [inline pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/BaseResource.java:270]: If we remove the `toString` call here, the `reason` may be null and be returned. We need to make sure that null is not returned here. The correct behavior is to return e.getMessage() when the `reason` is null.

**Review by RobertIndie — APPROVED** [review]

**Review by yuruguo — APPROVED** [review]

**Pull request merged**